code/game/objects/structures/training_machine.dm

```diff
@@ -94,14 +120,43 @@
 	to_chat(user, "<span class='notice'>You attach \the [attached_item] to the training device.</span>")
 	playsound(src, "rustle", 50, TRUE)
 
-/obj/structure/training_machine/proc/attach_item(target)
+
+/**
+ * Attach an item to the machine
+ *
+ * This proc technically works with any obj. Currently is only used for objects of type item/target and item/training_toolbox
+ * Will make the attached item appear visually on the machine
+ * Arguments
+ * * target - The object to attach
+ */
+/obj/structure/training_machine/proc/attach_item(obj/target)
 	remove_attached_item()
 	attached_item = target
 	attached_item.forceMove(src)
 	attached_item.vis_flags |= VIS_INHERIT_ID
 	vis_contents += attached_item
+	RegisterSignal(attached_item, COMSIG_PARENT_QDELETING, .proc/on_attached_delete)
 	handle_density()
-
+/**
+ * Called when the attached item is deleted.
+ *
+ * This should probably only happen through admins or explosions causing the machine to live but the attached item in the machines
+ * contents to somehow be deleted.
+ */
+/obj/structure/training_machine/proc/on_attached_delete()
+	message_admins("On attach delete called")
+	vis_contents -= attached_item
+	attached_item = null
+	UnregisterSignal(attached_item, COMSIG_PARENT_QDELETING)
```

Comment:
I'd put these into a proc or macro and call it by both `remove_attached_item()` and this proc.
Because there is behavior you are likely to miss adding to one that would make sense to be on the other, like `handle_density()`

Reply:
This is still valid, despite the original code is outdated.

Reply:
Went ahead and made remove_attached_item() call this proc